server/src/main/java/io/crate/expression/scalar/ArrayUpperFunction.java

```diff
@@ -171,20 +171,19 @@ public Query toQuery(Function parent, Function arrayLength, Context context) {
             // Storage of the multidimensional arrays is not supported.
             return null;
         }
-        DataType<?> elementType = ArrayType.unnest(arrayRef.valueType());
-        if (elementType.id() == ObjectType.ID || elementType.equals(DataTypes.GEO_SHAPE)) {
-            // No doc-values for these, can't utilize doc-value-count
-            return null;
-        }
         int cmpVal = cmpNumber.intValue();
-
```

Comment:
can we add a test similar to `IndexerTest.test_array_length_is_indexed` to check array sub-col behavior? 

Also, would be col to have 2 scenarios - array is leaf (the deepest) column and array is a sub-column of some object but in the middle of the tree and we have some deeper array-sub columns. 

My expectation would be that we index length on all levels and such can simplify approach in the second commit. see also my another comment.